tests/integration/test_charm_one_runner.py

```diff
@@ -189,6 +197,70 @@ async def test_reconcile_runners_with_lxd_storage_pool_failure(
 
 @pytest.mark.asyncio
 @pytest.mark.abort_on_fail
+async def test_wait_on_busy_runner_repo_check(
+    model: Model,
+    app_runner: Application,
+    github_repository: Repository,
+    runner_manager_github_client: GithubClient,
+) -> None:
+    """
+    arrange: A working application with no runners.
+    act:
+        1. Set dockerhub-mirror config and spawn one runner.
+        2. Dispatch a workflow.
+    assert:
+        1. registry-mirrors is setup in /etc/docker/daemon.json of runner.
+        2. Message about dockerhub_mirror appears in logs.
```

Comment:
May I ask how this checks that the busy_runners have been waited?

Reply:
I have changed the test.
The test was outdated.